File: src/main/java/com/dv/telegram/exception/CommandException.java
package com.dv.telegram.exception;

import java.util.List;

public class CommandException extends RuntimeException {

    private final List<String> errorMessages;

    public CommandException(String message) {
        super(message);
        errorMessages = List.of(message);
    }

    public CommandException(List<String> messages) {
        super(messages.get(0));
        errorMessages = messages;
    }

    public CommandException(String message, Throwable cause) {
        super(message, cause);
        errorMessages = List.of(message);
    }

    public CommandException(List<String> messages, Throwable cause) {
        super(messages.get(0), cause);
        errorMessages = messages;
    }

    public List<String> getErrorMessages() {
        return errorMessages;
    }
}

File: src/main/kotlin/com/dv/telegram/exception/CommandException.kt
package com.dv.telegram.exception

class CommandException : RuntimeException {
    val errorMessages: List<String>
    val useMarkdownInResponse: Boolean

    constructor(message: String, useMarkdownInResponse: Boolean = false) : super(message) {
        errorMessages = listOf(message)
        this.useMarkdownInResponse = useMarkdownInResponse
    }

    constructor(messages: List<String>, useMarkdownInResponse: Boolean = false) : super(messages[0]) {
        errorMessages = messages
        this.useMarkdownInResponse = useMarkdownInResponse
    }

    constructor(message: String, cause: Throwable?, useMarkdownInResponse: Boolean = false) : super(message, cause) {
        errorMessages = listOf(message)
        this.useMarkdownInResponse = useMarkdownInResponse
    }

    constructor(messages: List<String>, cause: Throwable?, useMarkdownInResponse: Boolean = false) : super(messages[0], cause) {
        errorMessages = messages
        this.useMarkdownInResponse = useMarkdownInResponse
    }
}
